agent/proxycfg/testing_tproxy.go

```diff
@@ -655,3 +655,117 @@ func TestConfigSnapshotTransparentProxyDestination(t testing.T) *ConfigSnapshot
 		},
 	})
 }
+
+func TestConfigSnapshotTransparentProxyDestinationHTTP(t testing.T) *ConfigSnapshot {
```

Comment:
This helper function is mostly identical to `TestConfigSnapshotTransparentProxyDestination` above aside from the protocol in the config entries - that function could probably be reused and the protocol/config entry modification parameterized to avoid copying the whole thing.

Reply:
This now have different data set. There is still something that could be done by extracting the entries and passing them in the call site but if that's ok I prefer to postpone that to another PR after the release.